contracts/tfi-factory/src/mock_querier.rs

```diff
@@ -1,18 +1,42 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, marker::PhantomData};
+
 use cosmwasm_std::testing::{MockApi, MockQuerier, MockStorage, MOCK_CONTRACT_ADDR};
 use cosmwasm_std::{
-    from_slice, to_binary, Coin, ContractResult, Empty, OwnedDeps, Querier, QuerierResult,
+    from_slice, to_binary, Addr, Coin, ContractResult, Empty, OwnedDeps, Querier, QuerierResult,
     QueryRequest, SystemError, SystemResult, WasmQuery,
 };
-use std::{collections::HashMap, marker::PhantomData};
+
 use tfi::asset::{AssetInfo, PairInfo};
 
+// Used for the create pairs migration admin
+pub const FACTORY_ADMIN: &str = "migrate_admin";
+
+// Copied here because this struct is non-exhaustive.
```

Comment:
Wasn't that actually changed in some recent version? :thinking:

Reply:
https://github.com/CosmWasm/cosmwasm/pull/1177

Reply:
Yes, but I need to set the admin. :-/ Need a `new_with_admin` helper, for tests.